manifests/config.pp

```diff
@@ -254,54 +254,91 @@
 #   string:  enable syslog logging of statistics
 #   values:  <tt>on</tt>, <tt>off</tt> or <tt><syslog facility></tt>
 #
+#  [*systemd*]
+#   boolean: Whether to use systemd mode or not
+#   values: <tt>true</tt>, <tt>false</tt>
+#   Default: <tt>OS compatible</tt>
+#
+#  [*expectation_sync*]
+#   String of Array of expectation sync. If value is a string it must
+#     be one of On or Off.
+#   values: <tt>On</tt>, <tt>Off</tt>, <tt>[Array]</tt>
```

Comment:
Could it be changed to be always an Array and do the logic in the template ? 
I think it would be easier when using this module, to be a parameter of just one type.

Reply:
Agree, we can use None as default instead of 'off' string.

Reply:
Using undef and an array always does not let to specify the 'On' keyword unless some dark magic is done using array values. I.E.: ['all'] could be translated to 'On', but in my opinion this is darker than the 'On', 'Off', [Array]

Reply:
Ok, this was just my opinion i guess i like too much to have single type vars, could you define expectation_sync to only accept Array or On|Off string values then? To avoid typos placing other string values

Reply:
Another option would be to have two different parameters, one for string and another for array, and check in the module class that they cannot be set at the same time.